src/lib/containers/Markdown.js

```diff
@@ -0,0 +1,588 @@
+                let row_values = row.values;
+                plot_data[j - 1].x.push(row_values[0]);
+                plot_data[j - 1].y.push(row_values[j]);
+            }
+        }
+        // error with clearing html - "" is not a function, wrapping in try/catch prevents the error
+        // although it doesn't catch it.
+        try {
+            elementBundle.element.innerHTML = ""; // clear formatting (e.g. <Synapse Widget></SynapseWidget>)
+        }
+        catch (e) {
+            console.log('element bundle error ', e);
+        }
+        // responsive plot
+        // https://plot.ly/javascript/responsive-fluid-layout/#responsive--fluid-layout
+        (function () {
+            var d3 = window.Plotly.d3;
+            var WIDTH_IN_PERCENT_OF_PARENT = 100, HEIGHT_IN_PERCENT_OF_PARENT = 75;
+            var gd3 = d3.select(elementBundle.element)
+                .append('div')
+                .style({
+                    width: WIDTH_IN_PERCENT_OF_PARENT + '%',
+                    'margin-left': (100 - WIDTH_IN_PERCENT_OF_PARENT) / 5 + '%',
+                    height: HEIGHT_IN_PERCENT_OF_PARENT + 'vh',
+                    'margin-top': (100 - HEIGHT_IN_PERCENT_OF_PARENT) / 5 + 'vh'
+                });
+            var gd = gd3.node();
+            window.Plotly.plot(gd, plot_data, layout, config);
+            window.onresize = function () {
+                window.Plotly.Plots.resize(gd);
+            };
+        })();
+    }
+
+    /**
+     * Get data for plotly
+     *
+     * @param {*} widgetparamsMapped
+     * @returns data corresponding to plotly widget
+     * @memberof Markdown
+     */
+    getPlotlyData(widgetparamsMapped) {
+
+        let raw_plot_data = {}
+
+        // step 1: get init query with maxRowsPerPage calculated
+        let queryRequest = {
+            concreteType: "org.sagebionetworks.repo.model.table.QueryBundleRequest",
+            entityId: this.state.ownerId,
+            query: {
+                isConsistent: false,
+                limit: 150,
+                partMask: 9,  // get query results and max rows per page
+                offset: 0,
+                sql: widgetparamsMapped.query
+            }
+        };
+
+        // Have to make two "sets" of calls for query, the first one tells us the maximum size per page of data
+        // we can get, the following uses that maximum and offsets to the appropriate location to get the data
+        // afterwards, the process repeats
+        this.props.getQueryTableResults(queryRequest, this.props.token).then(initData => {
+            let maxPageSize = initData.maxRowsPerPage
+            let queryCount = initData.queryResult.queryResults.rows.length
+            let totalQueryResults = queryCount
+
+            raw_plot_data = initData;
+
+            // Get the subsequent data, note- although the function calls itself, it runs
+            // iteratively due to the await
+            const getData = async (initGet) => {
```

Comment:
please remove initGet boolean.  if the queryCount is == maxPageSize, then we keep asking for data.  if < maxPageSize then this is the last page (which may or may not be the first page).